apps/api/src/app/admin/admin.service.ts

```diff
@@ -264,6 +264,8 @@ export class AdminService {
         }),
         this.prismaService.symbolProfile.count({ where })
       ]);
+      const assetProfiles = symbolProfileResult[0];
+      let count = symbolProfileResult[1];
```

Comment:
I've always wondered whether it can be solved more elegantly 😄

Reply:
```ts
const [assetProfiles, totalAssetCount] = await Promise.all([
	... , 
	...
]);
let count = totalAssetCount;
```
Im unsure if this would be more elegant than the other solution. Maybe a little bit more self explanatory than accessing the values only be their index.

Alternatively we could do something like this:
```ts
public async getMarketData(...): Promise<AdminMarketData>{
	const [assetProfiles, count] = await Promise.all([
		...,
		...
	]);
	adminMarketData.count = count;
	...
	adminMarketData.marketData = await Promise.all(...);

	if (presetId) { // this is btw not so elegant as well, we could get rid of one indentation here
    	if (presetId === 'ETF_WITHOUT_COUNTRIES') {
        	adminMarketData.marketData = adminMarketData.marketData.filter(...);
        } else if (presetId === 'ETF_WITHOUT_SECTORS') {
          adminMarketData.marketData = adminMarketData.marketData.filter(...);
        }

        adminMarketData.count = adminMarketData.marketData.length;
    }

	return adminMarketData;
}
```
But again, im not sure if this really makes it more elegant and if such a simple change should involve this refactoring.

Reply:
No, let's keep your initial solution